exporter/datadogexporter/stats.go

```diff
@@ -0,0 +1,79 @@
+// Copyright The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// +build !windows
+
+package datadogexporter
+
+import (
+	"github.com/DataDog/datadog-agent/pkg/trace/pb"
+	"github.com/DataDog/datadog-agent/pkg/trace/stats"
+)
+
+const (
+	statsBucketDuration int64 = 1e10 // 10 seconds
+)
+
+// ComputeAPMStats calculates the stats that should be submitted to APM about a given trace
+func ComputeAPMStats(tracePayload *pb.TracePayload, pushTime int64) *stats.Payload {
+
+	statsRawBuckets := make(map[int64]*stats.RawBucket)
+
+	bucketTS := pushTime - statsBucketDuration
+
+	for _, trace := range tracePayload.Traces {
+		spans := GetAnalyzedSpans(trace.Spans)
+		sublayers := stats.ComputeSublayers(trace.Spans)
+		for _, span := range spans {
+
+			// TODO: This is all hardcoded to assume 10s buckets for now
+
+			// Aggregate the span to a bucket by rounding its end timestamp to the closest bucket ts.
+			// E.g., for buckets of size 10, a span ends on 36 should be aggregated to the second bucket
```

Comment:
So, some of this got copy/pasta'd in from another repo. This comment is confusing and i'm changing it to reflect the current state of the code.  TL;DR ideally we'd keep multiple buckets in a buffer with a broad time frame, and only flush the absolute oldest bucket every 10s. This would allow traces that get reported late to have a higher chance of ending up in their appropriate bucket. This is how our vendor specific datadog-agent handles things, but it's big ask to port that logic over as it's very tightly coupled with other code in the agent. So this simplified approach is what we've chosen for now to at least ensure we accurately count hits, even if the hits are bucketed off by +- 10s or so.